src/components/PasswordBox.js

```diff
@@ -41,7 +41,7 @@ class PasswordBox extends Nimiq.Observable {
 
         this.$el.classList.toggle('hide-input', this.options.hideInput);
 
-        this._passwordInput = new PasswordInput(this.$el.querySelector('[password-input]'));
+        this._passwordInput = new PasswordInput(/** @type {HTMLElement} */(this.$el.querySelector('[password-input]')));
```

Comment:
Why is this change necessary? I thought with your changes `null | undefined` are also allowed?

Reply:
I think that it worked previously, is actually a typescript bug. `querySelector` returns `Element | null`, `PasswordInput` previously accepted `?HTMLElement` which is equivalent to `HTMLElement | null`, which should refuse a passed value of `Element | null` but didn't. Now it's `{?HTMLElement} [$el]`, which is equivalent to `HTMLElement | null | undefined`, which somehow trigger typscript to actually complain (i.e. it's now actually more restrictive), which is why a type assertion is now required. This is the way it should be imo.